components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -0,0 +1,184 @@
+    ) {
+        // Do initialization off of the main thread
+        scope.launch {
+            // Build Nimbus AppContext object to pass into initialize
+            var packageInfo: PackageInfo? = null
+            try {
+                packageInfo = context.packageManager.getPackageInfo(
+                    context.packageName, 0
+                )
+            } catch (e: PackageManager.NameNotFoundException) {
+                Log.log(Log.Priority.ERROR,
+                    LOG_TAG,
+                    message = "Could not retrieve package info for appBuild and appVersion"
+                )
+            }
+            // If `longVersionCode` is available, use it instead of `versionCode`
+            val experimentContext = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                AppContext(
+                    appId = context.packageName,
+                    androidSdkVersion = Build.VERSION.SDK_INT.toString(),
+                    appBuild = packageInfo?.longVersionCode.toString(),
+                    appVersion = packageInfo?.versionName,
+                    architecture = Build.SUPPORTED_ABIS[0],
+                    debugTag = null,
+                    deviceManufacturer = Build.MANUFACTURER,
+                    deviceModel = Build.MODEL,
+                    locale = getLocaleTag(),
+                    os = "Android",
+                    osVersion = Build.VERSION.RELEASE)
+            } else {
+                // ("VERSION.SDK_INT < P")
+                @Suppress("DEPRECATION")
+                AppContext(
+                    appId = context.packageName,
+                    androidSdkVersion = Build.VERSION.SDK_INT.toString(),
+                    appBuild = packageInfo?.versionCode.toString(),
+                    appVersion = packageInfo?.versionName,
+                    architecture = Build.SUPPORTED_ABIS[0],
+                    debugTag = null,
+                    deviceManufacturer = Build.MANUFACTURER,
+                    deviceModel = Build.MODEL,
+                    locale = getLocaleTag(),
+                    os = "Android",
+                    osVersion = Build.VERSION.RELEASE)
+            }
+
+            // Build a File object to represent the data directory for Nimbus data
+            dataDir = File(context.applicationInfo.dataDir, NIMBUS_DATA_DIR)
+
+            // Initialize Nimbus
+            experiments = Experiments(collectionName, experimentContext, dataDir.path, null)
+
+            // Get experiments
+            val activeExperiments = experiments.getActiveExperiments()
+
+            // Call Glean.setExperimentActive() for each active experiment
+            activeExperiments.forEach {
+                Glean.setExperimentActive(it.userFacingName, it.branchSlug)
+            }
+            // Note, we cannot call setExperimentInactive unless we track the state of the
+            // experiments in this component and persist the info, since Glean doesn't persist
```

Comment:
Does "this component" mean the code here in a-c, or the persistence layer of the underlying nimbus-sdk?

Reply:
Right now it would mean "in the Android-Component" because that's where Glean is integrated.  Once we migrate to using the Glean Rust API, when it is available, I believe everything we need is already available/persisted.  

Right now, the component only checks for experiments on startup (by-design), and since the Glean-experiments API doesn't persist the experiment info (it has to be set each run), the only need to call `setExperimentInactive` would be if we were currently enrolled in an experiment, and then in the same application run got an update from nimbus that says we were no longer enrolled in that experiment.

If we later add periodically checking for updates, then we will need to be sure an call the `setExperimentInactive` function correctly.